contracts/ColonyFunding.sol

```diff
@@ -244,17 +240,88 @@ contract ColonyFunding is ColonyStorage {
     emit RewardPayoutCycleEnded(_payoutId);
   }
 
-  function getRewardPayoutInfo(uint256 _payoutId) public view returns (bytes32, uint256, uint256, address, uint256) {
+  function getRewardPayoutInfo(uint256 _payoutId) public view returns (bytes32, uint256, uint256, uint256, address, uint256) {
     RewardPayoutCycle memory rewardPayoutInfo = rewardPayoutCycles[_payoutId];
     return (
       rewardPayoutInfo.reputationState,
+      rewardPayoutInfo.colonyWideReputation,
       rewardPayoutInfo.totalTokens,
       rewardPayoutInfo.amount,
       rewardPayoutInfo.tokenAddress,
       rewardPayoutInfo.blockTimestamp
     );
   }
 
+  function checkReputation(
+    bytes32 rootHash,
+    uint256 skillId,
+    address userAddress,
+    bytes key,
+    bytes value,
+    uint256 branchMask,
+    bytes32[] siblings
+  ) internal returns (uint256)
+  {
+    bytes32 impliedRoot = getImpliedRoot(key, value, branchMask, siblings);
+    require(rootHash == impliedRoot, "colony-reputation-invalid-root-hash");
+
+    uint256 reputationValue;
+    address keyColonyAddress;
+    uint256 keySkill;
+    address keyUserAddress;
+
+    assembly {
+      reputationValue := mload(add(value, 32))
+      keyColonyAddress := mload(add(key, 20))
+      keySkill := mload(add(key, 52))
+      keyUserAddress := mload(add(key, 72))
+    }
+
+    require(keyColonyAddress == address(this), "colony-reputation-invalid-colony-address");
+    require(keySkill == skillId, "colony-reputation-invalid-skill-id");
```

Comment:
There are no tests trying to check this require, or the next one. I could live without this one being tested (the worst thing that happens - where this is currently used, at least - is that they supply a different skill, which *must* have either the same or less reputation than they're entitled to claim against, so they're shorting themselves.

If the next require were accidentally removed somehow, then people would be able to claim against reputation held by another user, which seems bad...